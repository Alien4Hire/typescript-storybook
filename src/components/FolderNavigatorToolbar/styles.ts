import { Box, Button } from '@mui/material';

import { styled } from '@mui/material/styles';

export const Divider = styled('div')(({ theme }) => ({
  borderRight: `1px solid ${theme.palette.gray.medium}`,
  height: '34px',
  margin: 'auto',
}));

export const Label = styled('div')(({ theme }) => ({
  fontWeight: '600',
  fontSize: '15px',
  letterSpacing: '0.4px',
  color: theme.palette.gray.veryDark,
  fontFamily: "'Source Sans Pro', sans-serif",
}));

export const Wrapper = styled('div')(({ theme }) => ({
  position: 'relative',
  background: theme.palette.gray.veryLight,
  paddingLeft: '56px',
}));

export const FolderForm = styled('div')`
  display: flex;
  align-items: center;
`;

export const FolderAdd = styled(Box, {
  shouldForwardProp: (prop: string) => prop !== 'active',
})(({ active, theme }) => ({
  position: 'absolute',
  top: 0,
  bottom: 0,
  left: 0,
  right: 0,
  width: 'auto',
  padding: '2px',
  backgroundColor: theme.palette?.gray.veryLight,
  display: 'none',
  ...(active && {
    display: 'flex',
    alignItems: 'center',
  }),
}));

export const CsvModalBackdrop = styled(Box)({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: '#f1f1f1',
  borderRadius: '16px',
  padding: '32px 32px 70px',
});

export const UploadListItem = styled(Box)(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  padding: '16px 0px',
  borderBottom: `1px solid ${theme.palette.gray.light}`,
}));

export const UploadButton = styled(Button)({
  textTransform: 'none',
  width: 'fit-content',
  marginTop: '16px',
});

export const DownloadTemplateWrapper = styled(Box)({
  display: 'flex',
  alignItems: 'center',
  marginTop: '24px',
  cursor: 'pointer',
});
