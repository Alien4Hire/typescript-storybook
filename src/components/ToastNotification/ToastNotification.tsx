import React, { FunctionComponent as FC } from 'react';

import Snackbar from '@mui/material/Snackbar';
import MuiAlert, { AlertProps, AlertColor } from '@mui/material/Alert';

export type MlToastNotificationPropsT = {
  open: boolean;
  onClose?: () => void;
  message: string;
  type: AlertColor;
  testId?: string | undefined;
};

const Alert = React.forwardRef<HTMLDivElement, AlertProps>(function Alert(
  props,
  ref
) {
  return <MuiAlert elevation={6} ref={ref} variant="filled" {...props} />;
});

export const ToastNotification: FC<MlToastNotificationPropsT> = ({
  open,
  onClose = () => {},
  message,
  type,
  testId,
}) => {
  const handleClose = (
    event?: React.SyntheticEvent | Event,
    reason?: string
  ) => {
    if (reason === 'clickaway') {
      return;
    }
    onClose();
  };

  return (
    <Snackbar
      anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
      open={open}
      autoHideDuration={6000}
      onClose={handleClose}
      data-testid={testId}
    >
      <Alert onClose={handleClose} severity={type} sx={{ width: '100%' }}>
        {message}
      </Alert>
    </Snackbar>
  );
};
