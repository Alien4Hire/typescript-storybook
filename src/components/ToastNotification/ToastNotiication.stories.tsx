import React from 'react';

import { ComponentStory, ComponentMeta } from '@storybook/react';

import Button from '@mui/material/Button';
import { ToastNotification } from './ToastNotification';
import Stack from '@mui/material/Stack';
import { AlertColor } from '@mui/material/Alert';

export default {
  title: 'Component/ToastNotification',
  component: ToastNotification,
} as ComponentMeta<typeof ToastNotification>;

const Template: ComponentStory<typeof ToastNotification> = () => {
  const [open, setOpen] = React.useState<boolean>(false);
  const [notificationType, setNotificationType] =
    React.useState<AlertColor>('success');

  const handleClick = (type: AlertColor) => {
    setOpen(true);
    setNotificationType(type);
  };

  const handleClose = () => {
    setNotificationType('success');
    setOpen(false);
  };

  return (
    <Stack spacing={2} sx={{ width: '100%' }}>
      <Button
        variant="outlined"
        color="primary"
        onClick={() => handleClick('success')}
      >
        CLICK TO OPEN A SUCCESS NOTIFICATION
      </Button>
      <Button
        variant="outlined"
        color="primary"
        onClick={() => handleClick('error')}
      >
        CLICK TO OPEN AN ERROR NOTIFICATION
      </Button>
      <Button
        variant="outlined"
        color="primary"
        onClick={() => handleClick('warning')}
      >
        CLICK TO OPEN A WARNING NOTIFICATION
      </Button>
      <Button
        variant="outlined"
        color="primary"
        onClick={() => handleClick('info')}
      >
        CLICK TO OPEN AN INFO NOTIFICATION
      </Button>
      <ToastNotification
        open={open}
        onClose={handleClose}
        message={`${notificationType} toast notification`}
        type={notificationType}
      />
    </Stack>
  );
};

export const ToastNotificationView = Template.bind({});
