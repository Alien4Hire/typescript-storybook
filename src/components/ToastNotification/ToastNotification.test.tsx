import { render, screen } from '@testing-library/react';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';
import { ToastNotification } from './ToastNotification';
import { theme } from '../../theme';
import '../../setupTests';

const baseProps = {
  open: true,
  onClose: () => {},
  message: 'Toast Notification',
};

const renderComponent = (customProps?: { testId: string }) => {
  const props = { ...baseProps, ...customProps };

  return render(
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={theme}>
        <ToastNotification {...props} type="success" />
      </ThemeProvider>
    </StyledEngineProvider>
  );
};

describe('<ToastNotification />', () => {
  it('renders without crashing', () => {
    const view = renderComponent();
    expect(view).toMatchSnapshot();
  });

  it('should have a message', () => {
    renderComponent();
    expect(screen.getByText(baseProps.message)).toBeInTheDocument();
  });

  it('should have a data-test-id', () => {
    const testId = 'test-toast-notification';
    renderComponent({ testId });
    const element = screen.getByTestId(testId);
    expect(element).toBeInTheDocument();
  });
});
