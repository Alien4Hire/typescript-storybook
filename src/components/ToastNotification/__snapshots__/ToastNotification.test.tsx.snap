// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToastNotification /> renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  background-color: #ffffff;
  color: rgba(0,0,0,0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.c0 {
  z-index: 1400;
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  left: 8px;
  right: 8px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 8px;
}

.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c7::-moz-focus-inner {
  border-style: none;
}

.c7.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c8 {
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0,0,0,0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  color: inherit;
  padding: 5px;
  font-size: 1.125rem;
}

.c8:hover {
  background-color: rgba(0,0,0,0.04);
}

.c8.Mui-disabled {
  background-color: transparent;
  color: rgba(0,0,0,0.26);
}

.c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: inherit;
}

.c9 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.25rem;
}

.c2 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.43;
  background-color: #2e7d32;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 6px 16px;
  color: #fff;
  background: #daf4d4;
  color: #080808;
  border: 1px solid #3b822e;
  width: 100%;
}

.c3 {
  margin-right: 12px;
  padding: 7px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 22px;
  opacity: 0.9;
}

.c5 {
  padding: 8px 0;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 4px 0 0 16px;
  margin-left: auto;
  margin-right: -8px;
}

@media (min-width:660px) {
  .c0 {
    top: 24px;
    left: 50%;
    right: auto;
    -webkit-transform: translateX(-50%);
    -ms-transform: translateX(-50%);
    transform: translateX(-50%);
  }
}

@media print {
  .c7 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c8:hover {
    background-color: transparent;
  }
}

<body>
    <div>
      <div
        class="c0 MuiSnackbar-root MuiSnackbar-anchorOriginTopCenter"
        role="presentation"
      >
        <div
          class="c1 MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation6 c2 MuiAlert-root MuiAlert-filledSuccess MuiAlert-filled"
          direction="down"
          role="alert"
          style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            class="c3 MuiAlert-icon"
          >
            <svg
              aria-hidden="true"
              class="c4 MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              data-testid="SuccessOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
              />
            </svg>
          </div>
          <div
            class="c5 MuiAlert-message"
          >
            Toast Notification
          </div>
          <div
            class="c6 MuiAlert-action"
          >
            <button
              aria-label="Close"
              class="c7 MuiButtonBase-root c8 MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
              tabindex="0"
              title="Close"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="c9 MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                data-testid="CloseIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c1 {
  background-color: #ffffff;
  color: rgba(0,0,0,0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.c0 {
  z-index: 1400;
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  left: 8px;
  right: 8px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 8px;
}

.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c7::-moz-focus-inner {
  border-style: none;
}

.c7.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c8 {
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0,0,0,0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  color: inherit;
  padding: 5px;
  font-size: 1.125rem;
}

.c8:hover {
  background-color: rgba(0,0,0,0.04);
}

.c8.Mui-disabled {
  background-color: transparent;
  color: rgba(0,0,0,0.26);
}

.c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: inherit;
}

.c9 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.25rem;
}

.c2 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.43;
  background-color: #2e7d32;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 6px 16px;
  color: #fff;
  background: #daf4d4;
  color: #080808;
  border: 1px solid #3b822e;
  width: 100%;
}

.c3 {
  margin-right: 12px;
  padding: 7px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 22px;
  opacity: 0.9;
}

.c5 {
  padding: 8px 0;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 4px 0 0 16px;
  margin-left: auto;
  margin-right: -8px;
}

@media (min-width:660px) {
  .c0 {
    top: 24px;
    left: 50%;
    right: auto;
    -webkit-transform: translateX(-50%);
    -ms-transform: translateX(-50%);
    transform: translateX(-50%);
  }
}

@media print {
  .c7 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c8:hover {
    background-color: transparent;
  }
}

<div>
    <div
      class="c0 MuiSnackbar-root MuiSnackbar-anchorOriginTopCenter"
      role="presentation"
    >
      <div
        class="c1 MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation6 c2 MuiAlert-root MuiAlert-filledSuccess MuiAlert-filled"
        direction="down"
        role="alert"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="c3 MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="c4 MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            data-testid="SuccessOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
            />
          </svg>
        </div>
        <div
          class="c5 MuiAlert-message"
        >
          Toast Notification
        </div>
        <div
          class="c6 MuiAlert-action"
        >
          <button
            aria-label="Close"
            class="c7 MuiButtonBase-root c8 MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
            tabindex="0"
            title="Close"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="c9 MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              data-testid="CloseIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
